app/main.py

```diff
@@ -1,34 +1,86 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False) -> None:
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+        if start == end:
+            self.decks = [Deck(*start), ]
+        if start[0] == end[0]:
+            self.decks = [Deck(start[0], i)
+                          for i in range(start[1], end[1] + 1)]
+        if start[1] == end[1]:
+            self.decks = [Deck(i, start[1])
+                          for i in range(start[0], end[0] + 1)]
 
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def get_deck(self, row: int, column: int) -> Deck:
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
+
+    def fire(self, target: Deck) -> str:
+        target.is_alive = False
+        if sum([deck.is_alive for deck in self.decks]) == 0:
+            self.is_drowned = True
+            return "Sunk!"
+        return "Hit!"
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
```

Comment:
here I must use List[tuple] or not? I think not, we dont now about incomming data and we just not chek it in this task